src/boot/bootctl.c

```diff
@@ -251,6 +259,30 @@ static int get_file_version(int fd, char **v) {
         return r;
 }
 
+static const char *efi_arch(void) {
+#ifdef __x86_64__
+        _cleanup_free_ char *platform_size = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/firmware/efi/fw_platform_size", &platform_size);
+        if (r == -ENOENT)
+                return EFI_MACHINE_TYPE_NAME;
+        if (r < 0) {
+                log_error_errno(r, "Error reading EFI firmware platform size: %m");
+                return EFI_MACHINE_TYPE_NAME;
+        }
+
+        if (streq(platform_size, "64"))
+                return EFI_MACHINE_TYPE_NAME;
+        if (streq(platform_size, "32"))
+                return EFI_MACHINE_TYPE_ALT_NAME;
+
+        log_warning("Unknown EFI platform size: %s", platform_size);
```

Comment:
for warnings we usually say what we are doing now, i.e. suffix with ", ignoring" or so. But in this case we should probably say ", using the default word size"